packages/nx/src/command-line/explain.ts

```diff
@@ -0,0 +1,94 @@
+import * as chalk from 'chalk';
+import { join, relative } from 'path';
+import { nxConfigExplainer } from '../explain/config-explainers/nx-json';
+import { workspaceRoot } from '../utils/app-root';
+import { readJsonFile } from '../utils/fileutils';
+import { output } from '../utils/output';
+
+export interface ExplainArgs {
+  config: string;
```

Comment:
What are your thoughts on switching to `--file`? Not too strongly for one or the other.

Reply:
The tricky thing with being strict about this being a file path, is that we wouldn't really be able to allow users to explain the exact string that is in their extends:

E.g. nx.json

```json
{
  "extends": "nx/presets/core.json"
}
```

`nx/presets/core.json` is _not_ a valid file path for the user, it is a string which can successfully be resolved to a file path by node's module algorithm.

Therefore the user would have to do:

```sh
npx nx explain ./node_modules/nx/presets/core.json
```

In the case of third party/Nrwl presets, and that didn't feel as user friendly as just allowing them to pass any string which could be resolved by Node - therefore I went for "config" as the name